lib/api/bucketPut.js

```diff
@@ -41,7 +41,7 @@ export default function bucketPut(authInfo, request, locationConstraint, log,
     }
     // - AWS JS SDK sends a request with locationConstraint us-east-1
     // if no locationConstraint provided.
-    if (locationConstraint && config.locationConstraints &&
+    if (locationConstraint && !config.regions && config.locationConstraints &&
```

Comment:
And as just discussed, if we have the env variable "multiple" set we should make sure we have both locationConstraints and restEndpoints and otherwise throw an error at the config parsing stage.